pkg/planner/core/task.go

```diff
@@ -41,22 +41,8 @@ import (
 	"go.uber.org/zap"
 )
 
-var (
-	_ task = &copTask{}
-	_ task = &rootTask{}
-	_ task = &mppTask{}
-)
-
-// task is a new version of `PhysicalPlanInfo`. It stores cost information for a task.
-// A task may be CopTask, RootTask, MPPTaskMeta or a ParallelTask.
-type task interface {
-	count() float64
-	copy() task
-	plan() PhysicalPlan
-	invalid() bool
-	convertToRootTask(ctx PlanContext) *rootTask
-	MemoryUsage() int64
-}
+var _ Task = &copTask{}
+var _ Task = &mppTask{}
```

Comment:
here we just move single rootTask struct out of this file temproary, because the task interface defined below has used the *rootTask.

in next several pull requests, all the task type here, will be moved out into one same task pkg place, dependent of core pkg